### ISSUE TITLE
Arrow displays in SFML port where text should be

### ISSUE THREAD

**Issue by Quintus** [Secretchronicles/TSC#503]
In the SFML port, there are arrows displayed where text (especially points) should be. Screenshot:

![out](https://cloud.githubusercontent.com/assets/156358/12043202/00f7c416-ae83-11e5-99c6-6e449b9ca193.png)

This might be related to [this unported call to glPixelStorei](https://github.com/Secretchronicles/TSC/blob/f1a7838e6265ac2fb154ec24cfeaf778dd39c561/tsc/src/video/video.cpp#L1152-L1153).

Valete,
Quintus

**Comment by lambda-11235** [comment]
I don't think its glPixelStorei. Look at video/font.cpp line 121. Text rendering hasn't been ported to SFML yet.

**Comment by lambda-11235** [comment]
I got text to render, but my method substantially slows down the game when rendered. This causes stuttering when collecting gems. Currently, I have it so that the text is converted from a sf::Text to a sf::Image, and then converted to a cGL_Surface. This conversion is evidently a lot slower than drawing the text to screen. Any suggestions?

I've already done several changes to the code. Should I post my code somewhere?

**Comment by Quintus** [comment]
Taran Lynn <notifications@github.com> writes:

> I got text to render, but my method substantially slows down the game
> when rendered. This causes stuttering when collecting gems. Currently,
> I have it so that the text is converted from a sf::Text to a
> sf::Image, and then converted to a cGL_Surface. This conversion is
> evidently a lot slower than drawing the text to screen. Any
> suggestions?

We could incorporate SFML’s graphics module and switch to
sf::RenderWindow. It requires some cautions if used with our raw OpenGL
(see
http://www.sfml-dev.org/tutorials/2.3/window-opengl.php#using-opengl-together-with-the-graphics-module),
but this doesn’t look impossible. It would give us the ability to render
sf::Text directly with SFML’s own methods. Plus, using that was planned
in the future anyway.

> I've already done several changes to the code. Should I post my code
> somewhere?

For those pieces of code that you don’t file as a PR here, it is
absolutely okay to publish them in your own Git repository for
conservation. You can post comments on them to the mailinglist as the
TSC tracker probably wouldn’t be the appropriate place for code in other
repositories, but there’s still a relation to the TSC project.

Vale,
Quintus

-- 
#!/sbin/quintus
Blog: http://www.guelkerdev.de

GnuPG key: F1D8799FBCC8BC4F

**Comment by Quintus** [comment]
Culprit found. Well. Erm. I’m sorry.

https://github.com/Secretchronicles/TSC/blob/9cb251766520f17a9b865331a6fb23d3febfb83d/tsc/src/video/font.cpp#L120-L121

Now that I see it I remember having added that. Ouch.

Vale,
Quintus

**Comment by Quintus** [comment]
Depends on #512

**Comment by Quintus** [comment]
Current tip of `feature-sfml-port2` has `sf::Window` already replaced with `sf::RenderWindow` so use of SFML’s `sf::Text` is now possible. However, please review my explanations in #512 before starting.

Vale,
Quintus

**Comment by Quintus** [comment]
Resolved with implementation of #512.

Valete,
Quintus
